File: src/main/java/com/ddd/trainings/idea/domain/IdeaId.java
package com.ddd.trainings.idea.domain;

import static java.util.UUID.randomUUID;

import com.ddd.utils.BaseId;
import lombok.EqualsAndHashCode;
import lombok.Getter;

@EqualsAndHashCode(of = "value", callSuper = false)
public class IdeaId extends BaseId {

  @Getter
  private final String value;

  public IdeaId(String value) {
    this.value = value;
  }

  public static IdeaId create() {
    return new IdeaId(randomUUID().toString());
  }

  public static IdeaId from(String ideaId) {
    return new IdeaId(ideaId);
  }
}

File: src/Ddd/Trainings/Idea/Domain/IdeaId.cs
using System;
using Ddd.Utils;

namespace Ddd.Trainings.Idea.Domain;

public class IdeaId(string value) : BaseId
{
    private readonly string _value = value;

    public static IdeaId Create()
    {
        return new IdeaId(Guid.NewGuid().ToString());
    }

    public static IdeaId From(string ideaId)
    {
        return new IdeaId(ideaId);
    }

  public override bool Equals(object? obj)
  {
    return obj is IdeaId id &&
           _value == id._value;
  }

  public override string GetValue()
    {
        return _value;
    }

  public override int GetHashCode()
  {
    return HashCode.Combine(_value);
  }
} 